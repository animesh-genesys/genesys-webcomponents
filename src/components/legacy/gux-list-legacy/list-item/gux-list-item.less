@import (reference) '../../../../style/color.less';

// Variables part

:host {
  &:focus,
  &:active,
  &:hover:not([disabled]) {
    .gux-list-item {
      color: @gux-type-dark;
      background: @gux-blue-60;
    }
  }
}

:host(:focus),
:host(:active),
:host(:hover:not([disabled])) {
  .gux-list-item {
    color: @gux-type-dark;
    background: @gux-blue-60;
  }
}

:host([disabled]) {
  .gux-list-item {
    pointer-events: none;
    cursor: default;
    opacity: 0.5;
  }
}

.gux-list-item {
  display: flex;
  height: max-content;
  padding: 0 16px;
  line-height: 32px;
  word-wrap: break-word;
  cursor: pointer;

  .gux-text {
    width: 100%;
  }
}
