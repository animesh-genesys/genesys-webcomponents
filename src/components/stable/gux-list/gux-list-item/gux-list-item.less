@import (reference) '../../../../style/color.less';
@import (reference) '../../../../style/mixins.less';

:host {
  width: 100%;
  outline: none;
}

:host([disabled]) {
  pointer-events: none;
}

::slotted(gux-icon) {
  width: 16px;
  height: 16px;
  margin-right: 8px;
  vertical-align: middle;
}

button {
  all: unset;
  box-sizing: border-box;
  width: 100%;
  min-height: 32px;
  padding: 8px 16px;
  color: @gux-black-50;
  word-wrap: break-word;
  cursor: pointer;
  border: none;
  outline: none;

  &:focus:not(:disabled) {
    color: @gux-grey-100;
    background: @gux-blue-60;
  }

  &:hover:not(:disabled) {
    color: @gux-grey-100;
    background: @gux-blue-50;
  }

  &:active:not(:disabled) {
    color: @gux-grey-100;
    background: @gux-blue-40;
  }

  &:disabled {
    color: fade(@gux-black-50, 50%);
    cursor: default;
  }
}
