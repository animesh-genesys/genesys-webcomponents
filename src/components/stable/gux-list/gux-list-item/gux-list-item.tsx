import { Component, Element, h, Host, JSX, Listen, Prop } from '@stencil/core';

import { getClosestElement } from '../../../../utils/dom/get-closest-element';

@Component({
  styleUrl: 'gux-list-item.less',
  tag: 'gux-list-item',
  shadow: {
    delegatesFocus: true
  }
})
export class GuxListItem {
  @Element()
  root: HTMLGuxListItemElement;

  @Prop()
  disabled: boolean = false;

  @Listen('mouseup')
  onMouseup(): void {
    this.focusParentList();
  }

  @Listen('mouseover')
  onMouseover(): void {
    this.focusParentList();
  }

  private focusParentList(): void {
    const parentList = getClosestElement('gux-list', this.root) as HTMLElement;

    if (parentList && parentList.shadowRoot.activeElement === null) {
      this.root.blur();
      parentList.focus();
    }
  }

  render(): JSX.Element {
    return (
      <Host role="listitem">
        <button type="button" tabIndex={-1} disabled={this.disabled}>
          <slot></slot>
        </button>
      </Host>
    ) as JSX.Element;
  }
}
