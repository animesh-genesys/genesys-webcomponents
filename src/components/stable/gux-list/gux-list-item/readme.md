# gux-list-item



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description | Type      | Default |
| ---------- | ---------- | ----------- | --------- | ------- |
| `disabled` | `disabled` |             | `boolean` | `false` |


## Dependencies

### Used by

 - [gux-time-picker-beta](../../../beta/gux-time-picker)

### Graph
```mermaid
graph TD;
  gux-time-picker-beta --> gux-list-item
  style gux-list-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
