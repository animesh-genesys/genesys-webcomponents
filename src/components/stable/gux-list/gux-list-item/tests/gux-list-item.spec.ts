import { newSpecPage } from '@stencil/core/testing';
import { GuxListItem } from '../gux-list-item';

const components = [GuxListItem];
const language = 'en';

describe('gux-list-item', () => {
  it('should build', async () => {
    const page = await newSpecPage({
      components,
      html: `<gux-list-item>Item</gux-list-item>`,
      language
    });

    expect(page.rootInstance).toBeInstanceOf(GuxListItem);
  });

  describe('#render', () => {
    ['<gux-list-item>Item</gux-list-item>'].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
