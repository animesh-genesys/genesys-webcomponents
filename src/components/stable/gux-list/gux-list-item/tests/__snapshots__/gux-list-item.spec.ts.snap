// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-list-item #render should render component as expected (1) 1`] = `
<gux-list-item role="listitem">
  <mock:shadow-root>
    <button tabindex="-1" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Item
</gux-list-item>
`;
